internal/clients/aws.go

```diff
@@ -93,7 +104,13 @@ func SelectTerraformSetup(config *SetupConfig) terraform.SetupFn { // nolint:goc
 		}
 		ps.ClientMetadata = map[string]string{
 			keyAccountID: credCache.accountID,
+			keyPartition: "aws",
 		}
+
+		if pc.Spec.Endpoint != nil && pc.Spec.Endpoint.PartitionID != nil {
+			ps.ClientMetadata[keyPartition] = *pc.Spec.Endpoint.PartitionID
+		}
```

Comment:
this was also in my mind. I was a bit hesitant to source partitions from `github.com/aws/aws-sdk-go` as it has a deprecation notice [here](https://github.com/aws/aws-sdk-go#warning-this-sdk-is-in-maintenance-mode), stating

> Per that announcement, as of 7/31/2024, the SDK has entered maintenance mode. Going forward, we will limit releases to address critical bug fixes and security issues only. The SDK will not receive API updates for new or existing services, or be updated to support new regions.
Maintenance mode will last for 1 year. After 7/31/2025, the SDK will enter end-of-support, and no updates at all will be made. We recommend that you migrate to [AWS SDK for Go v2](https://aws.github.io/aws-sdk-go-v2/docs/). For additional details as well as information on how to migrate, please refer to the linked announcement.

Searched for an equivalent in `aws-sdk-go-v2` but relevant parts were moved to internal packages, (if I did not miss anything) I could not find anything exported for this.

probably for short term this won't be an issue, but I thought that we can implement something for now to enable the feature, and iterate for cheaper maintenance code.

Reply:
What Erhan described is pretty much exactly my original thinking when I first wrote this. It can be done with the go sdk v1, but not v2, and given the pending deprecation I didn't want to add a new dependency on v1.